src/components/dialog/index.js

```diff
@@ -53,85 +54,145 @@ class Dialog extends BaseComponent {
+
+    this.state = {
+      alignments: this.state.alignments,
+      deltaY: new Animated.Value(this.initialPosition)
+    };
+
+    if (props.dismissSwipeDirection) {
+      console.warn('Dialog component\'s prop \'dismissSwipeDirection\' is deprecated, please remove it');
+    }
+  }
 
   generateStyles() {
     this.styles = createStyles(this.props);
   }
 
-  onSwipe(gestureName) {
-    const {SWIPE_UP, SWIPE_DOWN} = swipeDirections;
-    const {dismissSwipeDirection} = this.props;
-
-    switch (gestureName) {
-      case SWIPE_UP:
-        if (dismissSwipeDirection === SWIPE_DIRECTIONS.UP) {
-          _.invoke(this.props, 'onDismiss');
-        }
-        break;
-      case SWIPE_DOWN:
-        if (dismissSwipeDirection === SWIPE_DIRECTIONS.DOWN) {
-          _.invoke(this.props, 'onDismiss');
-        }
-        break;
-      default:
-        break;
-    }
+  onDismiss = () => {
+    this.initPositions();
+    _.invoke(this.props, 'onDismiss');
   }
 
-  render() {
-    const {visible, overlayBackgroundColor, style, onDismiss, bottom, animationConfig, top} = this.getThemeProps();
-    const {alignments} = this.state;
-    const centerByDefault = _.isEmpty(alignments);
-    const config = {
-      velocityThreshold: 0.3,
-      directionalOffsetThreshold: 80,
-    };
+  initPositions() {
+    this.setState({
+      deltaY: new Animated.Value(this.initialPosition)
+    });
+  }
+
+  onModalShow = () => {
+    const {animationConfig} = this.getThemeProps();
+    const {deltaY} = this.state;
+ 
+    Animated.timing(deltaY, {
+      toValue: 0,
+      duration: _.get(animationConfig, 'duration', 280),
+      delay: _.get(animationConfig, 'delay', 200),
+      easing: _.get(animationConfig, 'easing', Easing.bezier(0.165, 0.84, 0.44, 1)),
+      useNativeDriver: _.get(animationConfig, 'useNativeDriver', true)
+    }).start();
+  }
+
+  renderContent() {
+    const {bottom} = this.getThemeProps();
     const bottomInsets = Constants.getSafeAreaInsets().paddingBottom;
+
+    return (
+      <TouchableWithoutFeedback>
+        <SafeAreaView style={{flexGrow: 1}}>
+          {this.props.children}
+          {Constants.isIphoneX && bottom && <View style={{height: bottomInsets}}/>}
+        </SafeAreaView>
+      </TouchableWithoutFeedback>
+    );
+  }
+
+  renderDraggableContainer() {
+    const {style, top} = this.getThemeProps();
+
+    return (
+      <PanGestureView
+        style={[this.styles.dialogContainer, style]}
+        direction={top && PanGestureView.directions.UP}
+        onDismiss={this.onDismiss}
+      >
+        {this.renderContent()}
+      </PanGestureView>
+    );
+  }
+
+  renderAnimationContainer() {
+    const {animationConfig, top} = this.getThemeProps();
+    const {alignments, deltaY} = this.state;
+    const centerByDefault = _.isEmpty(alignments);
     const animation = top ? AnimatableManager.presets.slideInDown : AnimatableManager.presets.slideInUp;
 
+    if (animationConfig && animationConfig.animation) {
```

Comment:
The two renders are almost identical...
maybe we can do the trick when we if-else just the type of container and then render it.